### ISSUE TITLE
permission error trying to access games.json in database folder

### ISSUE THREAD

**Issue by fpizzuta** [susanBuck/dwa15-fall2018#90]
i am getting the following:

[2018-11-03 17:09:56] production.ERROR: file_put_contents(/var/www/html/p3/database//games.json): failed to open stream: Permission denied {"[2018-11-03 17:09:56]

What is weird is that it was working for awhile and then stopped. I am guessing that when i pull updates this file maybe gets new permissions? How do i set the permissions without making it open to everyone?

Thanks

**Comment by jpersson74** [comment]
@fpizzuta- I had the same exact issue on my p2 project. Susan instructed me to ssh into the project directory/database and run chown www-data (your json filename).json and it allowed me to write to file. I now run it whenever there is a pull on production because the permissions do seem to get reverted whenever there is an update.

**Comment by susanBuck** [comment]
As @jpersson74 noted - any files/dirs the app needs to write to should be owned by the `www-data`user. This is how we set up the [storage and bootstrap/cache directory](https://github.com/susanBuck/dwa15-fall2018/blob/master/laravel/deploy-to-digital-ocean.md#permissions) when first deploying to prod.

If git is reverting your permission settings when you deploy new code, run `git config --global core.filemode` locally to see what your git config for filemode is. It should be `false` so filemode changes are ignored. 

This was a setting we made when [first setting up git](https://github.com/susanBuck/dwa15-fall2018/blob/master/servers-and-git/git-setup.md#initial-git-configuration) so it should be false, but it may have been changed?

If it's false and you're still seeing this issue after pulling code, let us know and we can dig deeper!
